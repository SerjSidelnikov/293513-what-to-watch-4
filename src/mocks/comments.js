export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,9`,
  },
];
