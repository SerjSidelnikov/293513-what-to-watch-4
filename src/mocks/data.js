export default {
  film: {
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: 2014,
  },
  movies: [
    `Fantastic Beasts`,
    `Bohemian Rhapsody`,
    `Macbeth`,
  ],
};
