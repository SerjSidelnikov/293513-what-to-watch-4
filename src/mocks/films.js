export default [
  {
    id: 1,
    title: `The Terminator`,
    poster: `https://picsum.photos/id/111/200/300`,
  },
  {
    id: 2,
    title: `Raging Bull`,
    poster: `https://picsum.photos/id/22/200/300`,
  },
  {
    id: 3,
    title: `Amelie`,
    poster: `https://picsum.photos/id/33/200/300`,
  },
  {
    id: 4,
    title: `Titanic`,
    poster: `https://picsum.photos/id/55/200/300`,
  },
  {
    id: 5,
    title: `Good Will Hunting`,
    poster: `https://picsum.photos/id/43/200/300`,
  },
  {
    id: 6,
    title: `Arrival`,
    poster: `https://picsum.photos/id/235/200/300`,
  },
  {
    id: 7,
    title: `Lost In Translation`,
    poster: `https://picsum.photos/id/122/200/300`,
  },
  {
    id: 8,
    title: `No Country For Old Men`,
    poster: `https://picsum.photos/id/109/200/300`,
  },
];
