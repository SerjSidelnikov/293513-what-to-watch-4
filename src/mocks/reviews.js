export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,9`,
  },
];
