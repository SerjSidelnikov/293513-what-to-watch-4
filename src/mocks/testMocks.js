export const promoItem = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: 2014,
};

export const film = {
  id: `1`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.360p.vp9.webm`,
  background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  cover: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `genre`,
  release: `2014`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  voiceCount: 240,
  duration: 99,
  director: `Wes Andreson`,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`,
  ],
  reviewIds: [`0`, `1`, `2`, `3`, `4`, `5`],
};

export const films = [
  {
    id: `1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.360p.vp9.webm`,
    background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    cover: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre`,
    release: `2014`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    voiceCount: 240,
    duration: 99,
    director: `Wes Andreson`,
    actors: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`,
    ],
    reviewIds: [`0`, `1`, `2`, `3`, `4`, `5`],
  },
  {
    id: `2`,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    background: `img/bohemian-rhapsody.jpg`,
    cover: `img/bohemian-rhapsody.jpg`,
    genre: `genre`,
    release: `2014`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    voiceCount: 240,
    duration: 99,
    director: `Wes Andreson`,
    actors: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`,
    ],
    reviewIds: [`0`, `1`, `2`, `3`, `4`, `5`],
  },
];

export const review = {
  id: `0`,
  author: `Kate Muir`,
  date: `December 24, 2016`,
  description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  rating: `8,9`,
};

export const reviews = [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,3`,
  },
];
