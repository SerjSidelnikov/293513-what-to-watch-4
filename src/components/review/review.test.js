import React from 'react';
import renderer from 'react-test-renderer';

import Review from './review';
import {review} from '../../mocks/testMocks';

describe(`Review`, () => {
  it(`Review rendered correctly`, () => {
    const tree = renderer.create(
        <Review review={review}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
