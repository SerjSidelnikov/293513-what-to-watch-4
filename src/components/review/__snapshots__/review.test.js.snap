// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Review rendered correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Kate Muir
        r
      </cite>
      <time
        className="review__date"
        dateTime="2016-12-24"
      >
        December 24, 2016
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    8,9
  </div>
</div>
`;
